### DISCUSSION TITLE
ee module Authentication  on Streamlit cloud

### DISCUSSION THREAD

**Discussion by leclab0** [giswqs/streamlit-geospatial#79] [Q&A] [answered]
Dear Prof Qiusheng Wu,

Thank you this great repo. I am also beginning to deploy my GEE code onto streamlit. 

However I encounter a problem with  the _ee.ServiceAccountCredentials(SERVICE_ACCOUNT, KEY)_ on strealit cloud.  as we have known the KEY argument is on json file but streamlit cloud ask to convert it to toml format and save it onto its secret. I manage to write in toml format and paste them into stremlit secret on cloud however it returns   _st.secrets has no attribute "endswith"_ . Did you also face this error in the past? I have struggled to solve this problem but still stcuk. I was wondering how you manage the authentification and initilization process on stremlit cloud. 

Thank you for you time. Look forward to hearing from you

Best regards,
Leclab staff

**Comment by giswqs** [comment, marked as answer]
Wrap the key with triple quotes, like this:

![image](https://user-images.githubusercontent.com/5016453/188054604-b4846c5f-2b48-46ee-9232-819cae1a3b4c.png)

**Reply by leclab0** [reply]
Aaaa unexpected, thank you very much
